composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/settings/DefaultEditSettingsComponent.kt

```diff
@@ -44,7 +45,7 @@ internal class DefaultEditSettingsComponent(
             GameSettings(
                 width = width,
                 height = height,
-                maxMines = maxMines,
+                maxMines = min(maxMines, width * height),
```

Comment:
It would be nice to also validate width and height to be minimum 2. And a maximum like 100? We can also use coerceAtLeast , coerceAtMost and coerceIn - a Kotlin idiomatic way.

However, does it make sense to have maxMines equal to width * height? Such a game is 100% fail from the first click. I suggest to validate as follows.

`maxMines.coerceIn(1, width * height - 1)`

What do you think?